### ISSUE TITLE
Non-Premium or not signed-in YouTube User Video Playback Issue on SmartTube SDK, Response code 403

### ISSUE THREAD

**Issue by raa-ju** [yuliskov/SmartTube#3648]
I have identified an issue where non-premium or not signed-in YouTube users are unable to play videos on the SmartTube SDK. However the premium users are able to play videos using smartTube SDK. The problem seems to stem from an authentication check that prevents non-premium users from accessing video playback. Below is a detailed description of the issue, a possible workaround, and a method to bypass this check. Please note that the provided solutions may not fully resolve the issue and additional steps may be required.

Steps to Reproduce:

Use a non-premium or not signed-in YouTube account.
Attempt to play a video on an application using the SmartTube SDK.
Observe that the video does not play as expected for non-premium or not signed-in users.
Expected Behavior:
Non-premium and not signed-in YouTube users should be able to play videos without any issues.

Actual Behavior:
Non-premium and not signed-in YouTube users are unable to play videos as expected. The videos either do not load or show an error message.

Possible Workaround:
A temporary workaround has been identified, which involves modifying the user agent string to mimic a premium YouTube account. Need to test it though.

Bypass Instructions:
For a more permanent solution, consider bypassing the Premium and sign-in checks by intercepting the authentication request and modifying the response. This approach involves advanced techniques such as network interception and response modification.

Steps to Bypass:
 - Use a network interception tool (e.g., Charles Proxy or Fiddler) to capture the authentication request from the YouTube app.
 - Identify the specific headers and tokens used for Premium and sign-in validation.
 - Modify the interception script to alter the response, effectively bypassing the Premium and sign-in checks.
 - Implement this script in your application, ensuring that it runs before the video playback request is made.

Note: These workarounds and bypass instructions may not fully resolve the issue for all users. Additional steps and further investigation may be required to address the underlying problem completely.

**Comment by raa-ju** [comment]
Premium users can play all videos, I tested with a premium youtube user and it worked for all type of videos using smartTube

**Comment by ArloG3300** [comment]
Use a non-premium or not signed-in YouTube account.
Attempt to play a video on an application using the SmartTube SDK.
Observe that the video does not play as expected for non-premium or not signed-in users.

_**I've signed out. And then further cleared cache and data in ST before unistalling and installing it again.
Remained signed out. I'm a non-premium user.
Nothing plays still with the 403 error. In the case of a Dolby Vision video I do see the icon in the upper right of my screen.
Capturing network traffic, ok I'm capable of that. Scripting is another thing.
Hopefully the devs. can work a solution_**_

Latest Fire TV Cube.


_****_

**Comment by ACheshirov** [comment]
Yeah, I got the Premium subscription because I have a free trial month. And just like that, magic happened. SmartTube started loading videos without even needing a restart...
